### PR TITLE
[codex] Add local thread store listing

### PR DESCRIPTION
**Pull request by wiltzius-openai**
Builds on top of #17659 

Move the filesystem + sqlite thread listing-related operations inside of a local ThreadStore implementation and call ThreadStore from the places that used to perform these filesystem/sqlite operations.

This is the first of a series of PRs that will implement the rest of the local ThreadStore.

Testing:
- added unit tests for the thread store implementation
- adjusted some unit tests in the realtime + personality packages whose callsites changed. Specifically I'm trying to hide ThreadMetadata inside of the local implementation and make ThreadMetadata a sqlite implementation detail concern rather than a public interface, preferring the more generate StoredThread interface instead
- added a corner case test for the personality migration package that wasn't covered by the existing test suite
- adjust the behavior of searched thread listing to run the existing local rollout repair/backfill pass _before_ querying SQLite results, so callers using ThreadStore::list_threads do not miss matches after a partial metadata warm-up

### CODE REVIEW

File: codex-rs/rollout/src/recorder.rs [hunk 1]
```diff
@@ -279,7 +279,7 @@ impl RolloutRecorder {
         let codex_home = config.codex_home();
         let state_db_ctx = state_db::get_state_db(config).await;
 
-        if search_term.is_some()
+        if search_term.is_none()
```
Comment (wiltzius-openai) [inline codex-rs/rollout/src/recorder.rs]: this early return was only safe for unsearched listing; for searched listing it could return a partially populated SQLite result before the existing rollout repair pass had a chance to backfill older matching threads

File: codex-rs/thread-store/src/types.rs [hunk 2]
```diff
@@ -135,6 +135,8 @@ pub struct ThreadPage {
 pub struct StoredThread {
     /// Thread id.
     pub thread_id: ThreadId,
+    /// Local rollout path when the backing store is filesystem-based.
+    pub rollout_path: Option<PathBuf>,
```
Comment (pakrym-oai) [inline codex-rs/thread-store/src/types.rs:139]: is this temporary?

File: codex-rs/core/src/realtime_context.rs [hunk 3]
```diff
@@ -117,33 +120,31 @@ pub(crate) async fn build_realtime_startup_context(
...
-            MAX_RECENT_THREADS,
-            /*anchor*/ None,
-            SortKey::UpdatedAt,
-            &[],
-            /*model_providers*/ None,
-            /*archived_only*/ false,
-            /*search_term*/ None,
-        )
+async fn load_recent_threads(sess: &Session) -> Vec<StoredThread> {
+    let config = sess.get_config().await;
```
Comment (pakrym-oai) [inline codex-rs/core/src/realtime_context.rs]: any reason not to toss LocalThreadStore onto sess.services?

File: codex-rs/thread-store/src/local/mod.rs [hunk 4]
```diff
@@ -0,0 +1,565 @@
...
+}
+
+async fn list_rollout_threads(
+    config: &RolloutConfig,
+    params: &ListThreadsParams,
+    cursor: Option<&codex_rollout::Cursor>,
+    sort_key: codex_rollout::ThreadSortKey,
+) -> ThreadStoreResult<codex_rollout::ThreadsPage> {
+    let page = if params.archived {
+        RolloutRecorder::list_archived_threads(
```
Comment (pakrym-oai) [inline codex-rs/thread-store/src/local/mod.rs:148]: maybe worth inlining later separately

File: codex-rs/thread-store/src/types.rs (hunk 2 above)
Comment (wiltzius-openai) [reply codex-rs/thread-store/src/types.rs:139]: It's temporary in the sense that an optional path is necessary in the ThreadStore until we make the path non-required in the AppServer protocol. The app server currently has a non-nullable path field when returning summaries of listed threads, so the ThreadStore sort of has to provide it.

I'm trying to do as much of this migration as possible underneath the current app server shape, but we will eventually need to change that layer too.

Do you have a preferred approach for how to handle that?

File: codex-rs/core/src/realtime_context.rs (hunk 3 above)
Comment (wiltzius-openai) [reply codex-rs/core/src/realtime_context.rs]: don't think so, can do

File: codex-rs/thread-store/src/local/mod.rs (hunk 4 above)
Comment (wiltzius-openai) [reply codex-rs/thread-store/src/local/mod.rs:148]: by inline do you mean moving the archived/unarchived distinction into a param on the rollout recorder methods?

Comment (pakrym-oai) [reply codex-rs/thread-store/src/local/mod.rs:148]: I mean getting rid of RolloutRecorder and having methods be native on LocalThreadStorage.

RolloutRecorder was the previous place where we were abstracting away rollout management.

File: codex-rs/thread-store/src/types.rs (hunk 2 above)
Comment (pakrym-oai) [reply codex-rs/thread-store/src/types.rs:139]: I think keeping it here for a bit is fine especially considering rollout paths are all over app-server.

File: codex-rs/thread-store/src/local/mod.rs (hunk 4 above)
Comment (wiltzius-openai) [reply codex-rs/thread-store/src/local/mod.rs:148]: ah I see. Yes we can do that over time!

**Review by pakrym-oai — APPROVED** [review]
